core/src/main/java/io/undertow/predicate/SecurePredicate.java

```diff
@@ -33,7 +36,10 @@ public class SecurePredicate implements Predicate {
 
     @Override
     public boolean resolve(HttpServerExchange value) {
-        return value.getRequestScheme().equals("https");
+        final SSLSession session = value.getConnection().getSslSession();
+        final SSLSessionInfo info = value.getConnection().getSslSessionInfo();
+        return (session != null && session.isValid())
+                || (info != null && info.getSSLSession() != null && info.getSSLSession().isValid());
```

Comment:
Do we need to check `SSLSessionInfo` as well? I thought checking `SSLSession` would be sufficient for this use case. Am I misunderstanding or missing anything?

Reply:
Why not both? ( I think in source, though not sure where, I found similar check)